src/braket/circuits/gate.py

```diff
@@ -135,21 +146,33 @@ def _to_openqasm(
             target (QubitSet): target qubit(s).
             serialization_properties (OpenQASMSerializationProperties): The serialization properties
                 to use while serializing the object to the IR representation.
+            control (Optional[QubitSet]): Control qubit(s).
+            control_state (Optional[ControlState]): Quantum state on which to control the operation.
+                Must be a binary sequence of same length as number of qubits in `control`. Will be
+                ignored if `control` is not present. May be represented as a string, list, or int.
+                For example "0101", [0, 1, 0, 1], 5 all represent controlling on qubits 0 and 2
+                being in the |0⟩ state and qubits 1 and 3 being in the |1⟩ state.
+                Default "1" * len(control).
 
         Returns:
             str: Representing the openqasm representation of the gate.
         """
         target_qubits = [serialization_properties.format_target(int(qubit)) for qubit in target]
+        control_prefix = ""
         if control:
             control_qubits = [
                 serialization_properties.format_target(int(qubit)) for qubit in control
             ]
-            control_prefix = (
-                "ctrl @ " if (num_control := len(control)) == 1 else f"ctrl({num_control}) @ "
-            )
+            control_state_list = _as_list(control_state, len(control))
+            for state, group in groupby(control_state_list):
+                modifier_name = "neg" * (not state) + "ctrl"
```

Comment:
Surprised this actually works

Reply:
multiplying string by int/bool is pretty standard python no?